src/hotspot/share/gc/serial/serialHeap.cpp

```diff
@@ -187,19 +187,26 @@ jint SerialHeap::initialize() {
 
   initialize_reserved_region(heap_rs);
 
-  ReservedSpace young_rs = heap_rs.first_part(MaxNewSize, GenAlignment);
-  ReservedSpace old_rs = heap_rs.last_part(MaxNewSize, GenAlignment);
-
   _rem_set = new CardTableRS(_reserved);
-  _rem_set->initialize(young_rs.base(), old_rs.base());
+
+  if (SwapSerialGCGenerations) {
+    ReservedSpace old_rs = heap_rs.first_part(MaxOldSize, GenAlignment);
```

Comment:
For my own edification: Why does the original ReservedHeapSpace use MaxNewSize for both partitions, and why do we change that to MaxOldSize for both partitions when we swap their locations?